src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1471,6 +1472,17 @@ func (k *kataAgent) handleEphemeralStorage(mounts []specs.Mount) ([]*grpc.Storag
 				dir_options = append(dir_options, fmt.Sprintf("%s=%d", fsGid, stat.Gid))
 			}
 
+			// parse size option
+			sizeOption, err := parseEmptyDirSize(origin_src)
+			if err != nil {
+				k.Logger().WithError(err).Errorf("failed to parse empty dir size for %s", origin_src)
+				return nil, err
+			}
+			if sizeOption != "" {
+				// FIXME: How about a large volume size larger than guest's memory size?
```

Comment:
Fail?

Reply:
If we're going to try to deduce this information and handle this non-OCI "feature", I think we should properly handle (or error as Marcel suggests).

Would we want to increase the amount of memory for the guest, regardless, if we are going to create a tmpfs for this emptydir (ie, in the event the emptyDir is backed be memory)?

Reply:
I think what'd make most sense here is to ensure it is then sized to be the sum of container memory limits? The challenge that we have, though, is that we do not necessarily know what the sum of container memory limits is at this point, and may need to continually update the value when we update sandbox resources.

sizeOption, err := parseEmptyDirSize
if != "" {
  if sizeOption > sandbox-size {
       sizeOption = sandbox-size
  }
} else { 
      sizeOption = sandbox-size  
}

dir_options = append(dir_options, sizeOption)

Reply:
This seems to be an ideal solution, and the volume's size will be limited to the sandbox size.